walkeeper/src/timeline.rs

```diff
@@ -483,4 +486,80 @@ impl Storage for FileStorage {
         }
         Ok(())
     }
+
+    fn truncate_wal(&mut self, server: &ServerInfo, end_pos: Lsn) -> Result<()> {
+        let partial;
+        const ZERO_BLOCK: &[u8] = &[0u8; XLOG_BLCKSZ];
+        let wal_seg_size = server.wal_seg_size as usize;
+        let ztli = server.ztli;
+
+        /* Extract WAL location for this block */
+        let mut xlogoff = end_pos.segment_offset(wal_seg_size) as usize;
+
+        /* Open file */
+        let mut segno = end_pos.segment_number(wal_seg_size);
+        // note: we basically don't support changing pg timeline
+        let wal_file_name = XLogFileName(server.tli, segno, wal_seg_size);
```

Comment:
It seems that we're no longer using `server.tli` after https://github.com/zenithdb/zenith/pull/829, so `PG_CLI` should be used here. I'll fix this after rebase.